core/xml.js

```diff
@@ -349,6 +349,8 @@ Blockly.Xml.domToWorkspace = function(xml, workspace) {
       }
     } else if (name == 'shadow') {
       goog.asserts.fail('Shadow block cannot be a top-level block.');
+    }  else if (name == 'variables') {
+      Blockly.Xml.domToVariables(xmlChild, workspace);
```

Comment:
Is it valid to have two variables tags?  If not, check that this is the first tag and throw an error otherwise.

You'll want to throw an error instead of an assert because asserts don't run in compiled blockly.  (Ignore the assert above; we'll fix it eventually)